examples/vue-wagmi/src/App.vue

```diff
@@ -16,7 +16,7 @@ if (!projectId) {
 }
 
 // 2. Create wagmiConfig
-const chains = [mainnet, arbitrum]
+const chains = [mainnet, arbitrum] as [Chain, ...Chain[]]
```

Comment:
This looks very odd, why do we need to do it this way? Will users of typescript also need to add this type assertion?

Reply:
This shouldn't really be necessary as `[Chain, ...Chain[]]` should only mean "a Chain array with at least an element" but seems there's something weird with wagmi forcing us to cast it.

I changed the way we type things for now for a `Chain[]` and added a check to force users to provide at least 1 chain to the config.

Reply:
On this note, why do we have `chains` parameter on Web3ModalOptions? Shouldn't we use the ones in wagmi config?